bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,309 @@
+            color=Color.from_rgb(
+                *mapping_of_kyu[int(kata_information['rank']['name'].replace(' kyu', ''))]
+            ),
+        )
+        kata_embed.add_field(name="Difficulty", value=kata_information['rank']['name'], inline=False)
+        return kata_embed
+
+    @staticmethod
+    def language_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'language embed' which displays all languages the kata supports.
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
+        """
+        languages = '\n'.join(map(str.title, kata_information['languages']))
+        language_embed = Embed(
+            title="Languages Supported",
+            description=f"```nim\n{languages}```",
+            color=Colours.python_blue,
+        )
+        return language_embed
+
+    @staticmethod
+    def tags_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'tags embed' which displays all the tags of the Kata.
+
+        Tags are what the Kata is about, and is another word for 'categories'
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
+        """
+        tags = '\n'.join(kata_information['tags'])
+        tags_embed = Embed(
+            title="Tags",
+            description=f"```nim\n{tags}```",
+            color=Colours.grass_green,
+        )
+        return tags_embed
+
+    @staticmethod
+    def miscellaneous_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'other information embed' which displays miscellaneous information about the kata.
+
+        This embed shows statistics like the total number of people who completed the kata,
+        the total number of stars of the Kata, etc.
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
+        """
+        miscellaneous_embed = Embed(
+            title="Other Information",
+            color=Colours.grass_green,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Score`",
+            value=f"**{kata_information['voteScore']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Stars`",
+            value=f"**{kata_information['totalStars']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Completed`",
+            value=f"**{kata_information['totalCompleted']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Attempts`",
+            value=f"**{kata_information['totalAttempts']}**",
+            inline=False,
+        )
+        return miscellaneous_embed
+
+    @staticmethod
+    def create_view(dropdown: InformationDropdown, link: str) -> ui.View:
+        """
+        Creates the 'view', which is essentially a name for the discord UI.
```

Comment:
"Discord UI" is also another made up name and does not make more sense in the grand-scheme of the Discord API.

```suggestion
        Creates the discord.py View for the Discord message components (dropdowns and buttons).
```